.devcontainer/Dockerfile

```diff
@@ -1,11 +1,9 @@
 FROM mcr.microsoft.com/vscode/devcontainers/java:11
 
-RUN apt-key adv --refresh-keys --keyserver keyserver.ubuntu.com\
-  && apt-get update && export DEBIAN_FRONTEND=noninteractive \
-  && apt-get -y install --no-install-recommends yarn
+# Set up nodesource, install node, yarn, fontconfig (for static viz), and Clojure
 
-RUN curl -fsSL https://deb.nodesource.com/setup_14.x | bash
-RUN apt-get update && apt-get -y install --no-install-recommends nodejs
-
-RUN curl -O https://download.clojure.org/install/linux-install-1.11.0.1100.sh \
-  && bash ./linux-install-1.11.0.1100.sh
+RUN ( curl -fsSL https://deb.nodesource.com/setup_14.x | bash ) \
```

Comment:
If this is combined into one very long line, then we won't benefit from Docker's caching of container layer. Is this intentional?

Reply:
It's all a tradeoff between caching vs. layer size. I moved the Clojure install to its own `RUN` to take advantage of layer caching when changing Clojure versions. Otherwise, most of the time I rebuild without cache more often than not. Installing nodesource, updating `apt`, etc., probably should be run frequently without cache to get the latest versions of those packages.